### PR TITLE
Add document describing requirements to file an issue

### PR DESCRIPTION
**Pull request by jcantrill**
cc @portante

This is a work in progress to document what we would like when people file issues.  I think some of this bit we could automate and I think maybe @lukas-vlcek was playing with a utility or script we can ref to get information.

### CODE REVIEW

**Comment by jcantrill** [comment]
cc @pweil- Tag you in for good measuer

File: docs/issues.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Filing An Issue
```
Comment (pweil-) [inline docs/issues.md]: can we add this as the GH template?

File: docs/issues.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+2. Openshift installer version
+3. Ansible inventory file
+4. Ansible installer logs
+
+### Logging Component Details
+**MAYBE WE CAN GRAB SOME OF THESE BY LABEL**
+1. Elasticsearch logs: `SOME NON-OC LOGS COMMAND`
+2. Other component logs: `oc logs $PODNAME`
+3. Fluentd spec: `oc get ds logging-fluentd -o yaml`
+4. Other component specs: `oc get dc $DCNAME -o yaml`
```
Comment (pweil-) [inline docs/issues.md]: don't you want the actual pod definitions so that you can see all defaults that get applied?

File: docs/issues.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+### Installer Details
+1. How was logging installed (e.g. Openshift Ansible/Installer, deployer)
+2. Openshift installer version
+3. Ansible inventory file
+4. Ansible installer logs
+
+### Logging Component Details
+**MAYBE WE CAN GRAB SOME OF THESE BY LABEL**
+1. Elasticsearch logs: `SOME NON-OC LOGS COMMAND`
```
Comment (pweil-) [inline docs/issues.md]: probably want a "don't forget to redact any sensitive info" reminder

File: docs/issues.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+2. Other component logs: `oc logs $PODNAME`
+3. Fluentd spec: `oc get ds logging-fluentd -o yaml`
+4. Other component specs: `oc get dc $DCNAME -o yaml`
+5. Configmaps: `oc get configmaps $CONFIGMAP -o yaml`
+6. Events: `oc get events $PODNAME -o yaml`
```
Comment (pweil-) [inline docs/issues.md]: Are events good enough?  How often do you need master or kubelet logs when debugging?

File: docs/issues.md (hunk 1 above)
Comment (jcantrill) [inline docs/issues.md]: We can.  I think we should also provide something as the BZ template too

File: docs/issues.md (hunk 2 above)
Comment (jcantrill) [inline docs/issues.md]: Is that different then the DC?  Should the resulting POD be what is defined in the DC/DS?  Do we want both?

File: docs/issues.md (hunk 4 above)
Comment (jcantrill) [inline docs/issues.md]: Master and node logs would be a good addition, though I have no concrete knowledge where to source those.

File: docs/issues.md (hunk 2 above)
Comment (pweil-) [inline docs/issues.md]: in the DC/RC the runtime environment is not fully defined.  Upon submission all the defaulting code for the pod is run.  Things like ImagePullPolicy, SecurityContext, etc are then all filled in.  Basically anything that runs in defaulters and pod admission plugins will not be viewed in the DC.  Having both may be valuable.

File: docs/issues.md (hunk 4 above)
Comment (pweil-) [inline docs/issues.md]: Usually `journalctl` or leave it up to the admin to figure it out

Comment (portante) [inline docs/issues.md]: Also include output from Wesley Hearns ES [health check](https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py) [1].

[1] https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py

File: docs/issues.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+4. Number of compute nodes and size (e.g. CPU, RAM)
+5. Logging Image versions
+
+**ADD NOTE HERE ON HOW TO RETRIEVE IMAGE VERSION FROM OCP AND ORIGIN**
+
+### Installer Details
+1. How was logging installed (e.g. Openshift Ansible/Installer, deployer)
+2. Openshift installer version
+3. Ansible inventory file
+4. Ansible installer logs
```
Comment (ewolinetz) [inline docs/issues.md]: We should also add deployer logs (if applicable)

**Comment by jcantrill** [comment]
It would be nice to package this tool into the image if it proves to be useful, but this alone could be a non-starter to doing that:
```
If you encounter OutOfMemoryExceptions, use the DIAG_JAVA_OPTS environment variable to set an -Xmx value greater than the standard 2g. Start with -Xmx4g and move up from there if necessary.
```
We would not have enough memory.

File: docs/issues.md (hunk 5 above)
Comment (jcantrill) [inline docs/issues.md]: added

File: docs/issues.md (hunk 3 above)
Comment (jcantrill) [inline docs/issues.md]: Added

File: docs/issues.md (hunk 2 above)
Comment (jcantrill) [inline docs/issues.md]: Updated

**Comment by richm** [comment]
We now have logging support in https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_health_checker
https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_health_checker/openshift_checks/logging

Should we focus our efforts to enhance that, or elsewhere?

**Comment by jcantrill** [comment]
[merge]

**Pull request merged**